build-system/compile/single-pass.js

```diff
@@ -164,7 +165,11 @@ exports.getBundleFlags = function(g) {
   Object.keys(g.packages)
     .sort()
     .forEach(function(pkg) {
-      srcs.push(pkg);
+      g.bundles[mainBundle].modules.push(pkg);
+      fs.outputFileSync(
+        `${g.tmp}/${pkg}`,
+        JSON.stringify(JSON.parse(readFile(pkg)), null, 4)
```

Comment:
`pkg-up` has an alternative called [`read-pkg-up`](https://github.com/sindresorhus/read-pkg-up) that also reads the contents of `package.json`. Not sure if using it can help simplify this code. Your call.

Reply:
Eh, it returns a weird object. Ideally, this wouldn't be necessary, but one of the `package.json` files has a duplicate field that causes Closure to barf. It tries to interpret it in strict mode (because esm is strict) and it's an error to have a duplicate field in strict.